elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ToElasticsearch.java

```diff
@@ -421,6 +432,58 @@ private static JsonObject convertRemoteMatchQuery(RemoteMatchQuery query) {
 		return wrapQueryForNestedIfRequired( query.getField(), matchQuery );
 	}
 
+	private static JsonObject convertRemoteSimpleQueryStringQuery(RemoteSimpleQueryStringQuery query) {
+		JsonBuilder.Object queryBuilder = JsonBuilder.object()
+				.addProperty( "query", query.getQuery() )
+				.addProperty( "default_operator", query.isMatchAll() ? "and" : "or" );
+
+		Set<String> analyzers = new HashSet<>();
+		String overridingRemoteAnalyzerName = null;
+
+		JsonArray fieldArray = new JsonArray();
+		for ( Field field : query.getFields() ) {
+			StringBuilder sb = new StringBuilder( field.getName() );
+			if ( field.getBoost() != DEFAULT_BOOST ) {
+				sb.append( BOOST_OPERATOR ).append( field.getBoost() );
+			}
+			fieldArray.add( sb.toString() );
+
+			String originalRemoteAnalyzerName = query.getOriginalRemoteAnalyzerReference().getAnalyzer().getName( field.getName() );
+			String queryRemoteAnalyzerName = query.getQueryRemoteAnalyzerReference().getAnalyzer().getName( field.getName() );
+			analyzers.add( queryRemoteAnalyzerName );
+			if ( !queryRemoteAnalyzerName.equals( originalRemoteAnalyzerName ) ) {
+				if ( overridingRemoteAnalyzerName == null ) {
+					overridingRemoteAnalyzerName = queryRemoteAnalyzerName;
+				}
+				else if ( !overridingRemoteAnalyzerName.equals( queryRemoteAnalyzerName ) ) {
+					throw LOG.unableToOverrideQueryAnalyzerWithMoreThanOneAnalyzersForSimpleQueryStringQueries(
+							Arrays.asList( overridingRemoteAnalyzerName, queryRemoteAnalyzerName ) );
+				}
+			}
+		}
+		// we always have at least one field defined
+		queryBuilder.add( "fields", fieldArray );
+
+		if ( overridingRemoteAnalyzerName != null ) {
+			if ( analyzers.size() == 1 ) {
+				queryBuilder.addProperty( "analyzer", overridingRemoteAnalyzerName );
+			}
+			else {
+				List<String> overridingAnalyzers = new ArrayList<>();
+				overridingAnalyzers.addAll( analyzers );
+				overridingAnalyzers.add( overridingRemoteAnalyzerName );
```

Comment:
From what I can see above, `overridingRemoteAnalyzerName` is already in the `analyzers` set, so we could just pass the set to the logger method (if you made the parameter a `Collection`, that is :grin: )

```java
			analyzers.add( queryRemoteAnalyzerName ); // Value added to the set here
 			if ( !queryRemoteAnalyzerName.equals( originalRemoteAnalyzerName ) ) {
 				if ( overridingRemoteAnalyzerName == null ) {
 					overridingRemoteAnalyzerName = queryRemoteAnalyzerName; // Same value put into overridingRemoteAnalyzerName here
 				}
```